taskchampion/taskchampion/src/storage/sqlite.rs

```diff
@@ -76,13 +76,24 @@ pub struct SqliteStorage {
 }
 
 impl SqliteStorage {
-    pub fn new<P: AsRef<Path>>(directory: P) -> anyhow::Result<SqliteStorage> {
-        // Ensure parent folder exists
-        std::fs::create_dir_all(&directory)?;
+    pub fn new<P: AsRef<Path>>(
+        directory: P,
+        create_if_missing: bool,
+    ) -> anyhow::Result<SqliteStorage> {
+        if create_if_missing {
+            // Ensure parent folder exists
+            std::fs::create_dir_all(&directory)?;
+        }
 
         // Open (or create) database
         let db_file = directory.as_ref().join("taskchampion.sqlite3");
-        let con = Connection::open(db_file)?;
+        let mut flags = OpenFlags::SQLITE_OPEN_READ_WRITE
+            | OpenFlags::SQLITE_OPEN_NO_MUTEX
+            | OpenFlags::SQLITE_OPEN_URI;
```

Comment:
Discussed in person: We could use a `OpenFlags::default()` to make sure we don't go out of sync with the default set of flags for `Connection::open_with_flags`